test/extended/cmd.sh

```diff
@@ -71,75 +72,69 @@ popd
 rm -rf $tmp
 
 # ensure a local-only image gets a docker image(not imagestream) reference created.
-[ "$(oc new-app test/scratchimage~https://github.com/openshift/ruby-hello-world.git --strategy=docker -o yaml |& tr '\n' ' ' | grep -E "from:\s+kind:\s+DockerImage\s+name:\s+test/scratchimage:latest\s+")" ]
+os::cmd::expect_success_and_text "oc new-app test/scratchimage~https://github.com/openshift/ruby-hello-world.git --strategy=docker -o yaml |& tr '\n' ' '" "from:\s+kind:\s+DockerImage\s+name:\s+test/scratchimage:latest\s+"
```

Comment:
you can run these commands you know :)
-o yaml outputs all the objects new-app is going to create in yaml format.  it's not a Template, it's just a list of objects that could be piped to oc create

we're looking at the buildconfig in that object to ensure its From reference is to the right thing.